amavisd-milter/mlfi.c

```diff
@@ -331,20 +338,23 @@ mlfi_connect(SMFICTX *ctx, char *hostname, _SOCK_ADDR * hostaddr)
     mlfi->mlfi_amasd = -1;
 
     /* Save connection informations */
-    if ((client_name = smfi_getsymval(ctx, "{client_name}")) != NULL) {
-	logmsg(LOG_INFO, "%s: Client name: %s", hostname, client_name);
-	if ((mlfi->mlfi_client_host = strdup(client_name)) == NULL) {
-	    logmsg(LOG_ERR, "%s: could not allocate memory", hostname);
-	    mlfi_setreply_tempfail(ctx);
-	    return SMFIS_TEMPFAIL;
-	}
+    if ((mlfi->mlfi_client_host = strdup(client_name)) == NULL) {
+	logmsg(LOG_ERR, "%s: could not allocate memory", client_host);
+	 mlfi_setreply_tempfail(ctx);
+	 return SMFIS_TEMPFAIL;
     }
-    if (hostname != NULL && *hostname != '\0' && mlfi->mlfi_client_host == NULL) {
-	if ((mlfi->mlfi_client_host = strdup(hostname)) == NULL) {
-	    logmsg(LOG_ERR, "%s: could not allocate memory", hostname);
-	    mlfi_setreply_tempfail(ctx);
-	    return SMFIS_TEMPFAIL;
-	}
+    client_name = smfi_getsymval(ctx, "{client_name}");
+    if (client_name == NULL || *client_name == '\0') {
+	client_name = client_host;
+    }
+    if (*client_name == '[') {
+	client_name = "unknown";
+    }
+    logqidmsg(mlfi, LOG_DEBUG, "client name: %s", client_name);
+    if ((mlfi->mlfi_client_name = strdup(client_name)) == NULL) {
```

Comment:
I guess you save the {client_name} macro's value here so that it can later be used for the Received header, but right now that still uses `mlfi_client_host`. Is that intentional or will you change the Received header code to use _name?

If the above is intentional: I'm not sure what values currently end up in the `_host` variable in the different cases and if they are want we want for spamassassin or not. I think that the old code worked fine for me when the hostname verification succeeded, so postfix probably sent the hostname via the parameter. When the verification failed it was somehow broken, but I don't remember how. I guess it might have sent NULL or an empty string, which means that your new code sets it to ''unknown". Do you know if that is what postfix does?

Reply:
Thanks, I forgot to use client_name in Received: header.

Reply:
Further down in the file there's a line where client_host is sent to amavis. It's around line 1191. Should this also be changed? I have no idea what that is used for and what the implication is.
```
amavisd_request(mlfi, "client_name", mlfi->mlfi_client_host)
```

Reply:
This is AM.PDP communication with the amavis daemon. Amavis uses it for logging and for some amavis checks. Spamassassins RDNS_NONE check uses the first trusted Received: header.